openeogeotrellis/load_stac.py

```diff
@@ -515,6 +509,68 @@ def intersects_temporally(interval) -> bool:
 
     band_names = metadata.band_names
 
+    if apply_lcfm_improvements or env.get(EVAL_ENV_KEY.LOAD_STAC_APPLY_LCFM_IMPROVEMENTS, False):
+        logger.info("applying LCFM resolution improvements")
+
+        requested_band_epgs = [epsgs for n, epsgs in band_epsgs.items() if n in band_names]
+        unique_epsgs = {epsg for epsgs in requested_band_epgs for epsg in epsgs}
+        requested_band_resolutions = [res for n, res in band_resolution.items() if n in band_names]
+
+        if len(unique_epsgs) == 1 and requested_band_resolutions:  # exact resolution
+            target_epsg = unique_epsgs.pop()
+            cell_width = cell_height = max(requested_band_resolutions)
```

Comment:
Is cell size rather than resolution so should be `min` instead ( ~ `layercatalog.py`); rename `band_resolution` and similars to `band_cell_size` etc?